homeassistant/components/scrape/config_flow.py

```diff
@@ -47,58 +58,110 @@
+    ),
     vol.Optional(CONF_USERNAME): TextSelector(),
     vol.Optional(CONF_PASSWORD): TextSelector(
         TextSelectorConfig(type=TextSelectorType.PASSWORD)
     ),
-    vol.Optional(CONF_HEADERS): ObjectSelector(),
-    vol.Optional(CONF_UNIT_OF_MEASUREMENT): TextSelector(),
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): NumberSelector(
+        NumberSelectorConfig(min=15, step=15, mode=NumberSelectorMode.BOX)
+    ),
+}
+
+SENSOR_SETUP_OPT = {
+    vol.Optional(CONF_ATTRIBUTE): TextSelector(),
+    vol.Optional(CONF_INDEX, default=0): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Required(CONF_SELECT): TextSelector(),
+    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
     vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorDeviceClass],
+            options=[cls.value for cls in SensorDeviceClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
     vol.Optional(CONF_STATE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorStateClass],
+            options=[cls.value for cls in SensorStateClass],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    ),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
+        SelectSelectorConfig(
+            options=[TEMP_CELSIUS, TEMP_FAHRENHEIT],
+            custom_value=True,
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
-    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
-    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
 }
 
-DATA_SCHEMA = vol.Schema({**SCHEMA_SETUP, **SCHEMA_OPT})
-DATA_SCHEMA_OPT = vol.Schema({**SCHEMA_OPT})
+SENSOR_SETUP = {
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+}
+
+
+def return_sensor_step(user_input: dict[str, Any]) -> str:
+    """Return sensor step."""
+    return "sensor"
+
+
+def validate_rest_setup(user_input: dict[str, Any]) -> dict[str, Any]:
+    """Validate rest setup."""
+    hass = async_get_hass()
+    rest_config = {
+        **dict(user_input),
+        CONF_SCAN_INTERVAL: int(user_input[CONF_SCAN_INTERVAL]),
+        CONF_TIMEOUT: int(user_input[CONF_TIMEOUT]),
+    }
+    try:
+        rest = create_rest_data_from_config(hass, rest_config)
+        asyncio.run_coroutine_threadsafe(rest.async_update(), hass.loop)
+    except Exception as err:
+        raise SchemaFlowError("resource_error") from err
+    return user_input
+
+
+DATA_SCHEMA_RESOURCE = vol.Schema(RESOURCE_SETUP)
+DATA_SCHEMA_SENSOR = vol.Schema({**SENSOR_SETUP, **SENSOR_SETUP_OPT})
```

Comment:
This seems to allow only a single sensor.
Is there a way to enable multiple sensors on a resource?

Reply:
That should not be too complicated but I was thinking better to do that in follow-up PR as for a reviewer that might be pretty complex to follow.

Reply:
I still think we should use a list.
Can we implement as a list with a single item, and implement adding/removing as a follow-up? 
I would love to have this included in 2022.12